client/blocks/get-apps/apps-badge.jsx

```diff
@@ -28,6 +28,10 @@ const APP_STORE_BADGE_URLS = {
 		defaultSrc: '/calypso/images/me/get-apps-ios-store.svg',
 		src: 'https://linkmaker.itunes.apple.com/assets/shared/badges/{localeSlug}/appstore-lrg.svg',
 		tracksEvent: 'calypso_app_download_ios_click',
+		storeLink: ( referrer ) =>
+			`https://apps.apple.com/app/apple-store/id335703880?pt=299112&ct=${ referrer }&mt=8`,
+		titleText: ( translate ) => translate( 'Download the WordPress iOS mobile app.' ),
```

Comment:
Seems this is the only place in this component where we need translations, so we can probably get rid of the `localize` HOC in favor of importing `translate` directly and use it here.

Also, I find weird when functions are named using nouns. A noun usually implies this is a simple property you can access with `object.property`. Can we maybe add a `get` prefix to explicitly denote this is a function you need to access with `object.getProperty()`?

```
import { translate } from 'i18n-calypso';

// ...

const APP_STORE_BADGE_URLS = {
  ios: {
    // ...
    getStoreLink: ( referrer ) =>
			`https://apps.apple.com/app/apple-store/id335703880?pt=299112&ct=${ referrer }&mt=8`,
    getTitleText: () => translate( 'Download the WordPress iOS mobile app.' ),
    getAltText: () => translate( 'Apple App Store download badge' ),
   // ...
}

// ...

export default connect( null, { recordTracksEvent } )( AppsBadge );
```

Reply:
>  localize HOC in favor of importing translate directly

Gotcha! 

>  I find weird when functions are named using noun

Agreed!  `storeLink` a string attribute at first, and I modify it to a function as an afterthought and didn't think about changing the name. Will modify it 👍

Reply:
> Seems this is the only place in this component where we need translations, so we can probably get rid of the localize HOC in favor of importing translate directly and use it here.

I fixed it.
But I don't quite understand what is the pro/cons of using `localize` HOC  vs importing translate... Importing `translate` always seems more handy... IS the only difference that with loxalize you can pass some arguments (ie. locale and nbFormat?)

Reply:
> But I don't quite understand what is the pro/cons of using `localize` HOC vs importing translate...

Check out the [package documentation](https://github.com/Automattic/wp-calypso/tree/master/packages/i18n-calypso#localize):

> The advantage of using a higher-order component instead of calling translate directly from the i18n-calypso module is that the latter does not properly account for change events which may be emitted by the state emitter object.

Basically, the `localize` HOC ensures components are [re-rendered whenever the locale changes](https://github.com/Automattic/wp-calypso/blob/d9b924732010fc94985703bc5af49cf80bdd2ee4/packages/i18n-calypso/src/localize.js#L33), so any translated string inside those components are updated.

I have to double check the side-effects of using `titleText: translate( 'Download the WordPress iOS mobile app.' )` since that might leave the translated string stuck with the previous locale when changing to a new one. 

We usually work around those side effects by placing the translated string inside a function like `getTitleText: () => translate( 'Download the WordPress iOS mobile app.' ),` since functions are evaluated in real-time (and thus use the current locale) while constants are evaluated only once on file load.

Reply:
Thanks for the clarification!